chore(docs): Add image CDN enablement docs (#35051)
gatsbyjs/gatsby: docs/docs/how-to/plugins-and-themes/creating-a-source-plugin.md

- When you click the "Open Preview" button in the CMS the `manifestId` in the URL should match the `manifestId` that the source plugin creates from that revision.
- The node manifests get written out in the `public` dir of your gatsby site, so you can check to manifests on your local disk `/public/__node-manifests/<sourcePluginName>/<manifestId>.json` or you can navigate directly to that piece of content `https://<your-domain>/__node-manifests/<sourcePluginName>/<manifestId>`

### Enabling Image CDN Support

Image CDN is a feature on Gatsby Cloud that provides edge network image processing by waiting to perform image processing until the very first user visit to a page. The processed image is then cached for super quick fetching on all subsequent user views. Enabling it will also speed up local development builds and production builds on other deployment platforms because images from your CMS or data source will only be downloaded if they are used in a created Gatsby page.

> Learn more [here](https://www.gatsbyjs.com/blog/image-cdn-lightning-fast-image-processing-for-gatsby-cloud/)

Author: This is much nicer - thanks!

```js
schema.buildObjectType({
  name: `YourImageAssetNodeType`,
  fields: {
    // .. your fields
  },
  interfaces: [`Node`, `RemoteFile`],
})
```

It is also recommended that you add a polyfill to provide support back through Gatsby v2. To do so, you simply wrap the `buildObjectType` call with the `addRemoteFilePolyfillInterface` polyfill like so.

```js
import { addRemoteFilePolyfillInterface } from "gatsby-plugin-utils/polyfill-remote-file"

addRemoteFilePolyfillInterface(
  schema.buildObjectType({
    name: `YourImageAssetNodeType`,
    fields: {
      // your fields
    },
    interfaces: [`Node`, `RemoteFile`],
  }),
  {
    schema, // schema you get as an argument from createSchemaCustomization
    actions, // the redux actions of gatsby also part of createSchemaCustomization
  }
)
```

Implementing the `RemoteFile` interface adds the correct fields to your new GraphQL type and adds the necessary resolvers to handle the type. `RemoteFile` holds the following properties:

- `mimeType: String!`
- `filename: String!`
- `filesize: Int`
- `width: Int`
- `height: Int`
- `publicUrl: String!`
- `resize(width: Int, height: Int, fit: enum): String`
- `gatsbyImage(args): String`

> You might notice that width, height, resize and `gatsbyImage` can be null. This is because the `RemoteFile` interface can also handle assets other than images, like PDF’s.

Reviewer: ```suggestion
You might notice that `width`, `height`, `resize`, and `gatsbyImage` can be `null`. This is because the `RemoteFile` interface can also handle assets other than images, like PDF’s.
```

Author: Is it standard in Gatsby docs to use the [Oxford comma](https://en.wikipedia.org/wiki/Serial_comma)?

> The string returned from `gatsbyImage` is intended to work seamlessly with [Gatsby Image Component](https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-plugin-image/#gatsbyimage) just like `gatsbyImageData` does.

Reviewer: ```suggestion
The string returned from `gatsbyImage` is intended to work seamlessly with [Gatsby Image Component](/docs/reference/built-in-components/gatsby-plugin-image/#gatsbyimage) just like `gatsbyImageData` does.
```

> ### `sourceNodes` Node API Additions

Reviewer: ```suggestion
#### `sourceNodes` node API additions
```

> When creating nodes, you must add some fields to the node itself to match what the `RemoteFile` interface used above when creating your new type expects. You may have guessed it already, you will need `url`, `mimeType`, `filename` as mandatory fields. When you have an image type, `width` and `height` are required as well. The optional fields are `placeholderUrl` and `filesize`. `placeholderUrl` will be the url used to generate blurred or dominant color placeholder so it should contain `%width%` and `%height%` url params if possible.

Reviewer: ```suggestion
When creating nodes, you must add some fields to the node itself to match what the `RemoteFile` interface expects. You will need `url`, `mimeType`, and `filename` as mandatory fields. When you have an image type, `width` and `height` are required as well. The optional fields are `placeholderUrl` and `filesize`. `placeholderUrl` will be the url used to generate blurred or dominant color placeholder so it should contain `%width%` and `%height%` url params if possible.
```

```js
const assetNode = {
  id: `an id`,
  internal: {
    type: `PrefixAsset`,
  },
  url: `${file.url}`,
  placeholderUrl: `${file.url}?w=%width%&h=%height%`,
  mimeType: file.contentType,
  filename: file.fileName,
  width: file.details?.image?.width,
  height: file.details?.image?.height,
}
```

> ### `onCreateDevServer` Node API

Reviewer: ```suggestion
#### `onCreateDevServer` node API
```

> You need one final polyfill to ensure that the dev server started with `gatsby develop` has the routes it needs to work with Image CDN.

Reviewer: ```suggestion
Add the polyfill `polyfillImageServiceDevRoutes` to ensure that the development server started with `gatsby develop` has the routes it needs to work with Image CDN.
```
